### DISCUSSION TITLE
OTAURL on Private Server not working

### DISCUSSION THREAD

**Discussion by AhmadNaeemBEL** [arendst/Tasmota#10455] [Q&A] [answered]
Hello,

I am trying to upgrade my Tasmota device from MQTT commands.
OTA is successful from OTAUrl on Tasmota official server.
But when I try to do OTA from my private server Tasmota gives the error.

14:36:10.019 MQT: stat/tasmota_027D83/UPGRADE = {"Upgrade":"Failed Server Did Not Report Size"}

I checked Tasmota official OTAUrl and my OTAUrl on postman following is the result.

This is the snip of my private OTAURl
![2](https://user-images.githubusercontent.com/76518366/103988685-5cec0e00-51b0-11eb-8be9-ebe605f25f49.PNG)

This is the snip of Tasmota OTAUrl.
![3](https://user-images.githubusercontent.com/76518366/103988706-61182b80-51b0-11eb-9f82-db5e6fca2b43.PNG)

Kindly help me resolve the issue.

Kind Regards

**Comment by Jason2866** [comment]
Try a different webserver. For example https://github.com/arendst/Tasmota/tree/development/tools/ota_server

**Comment by sfromis** [comment]
I'm seeing that message if the server responds with an error message (or did not include the header field with length). Check with a browser to see if it responds with a binary file.

**Reply by AhmadNaeemBEL** [reply]
Yes, it responds.

**Comment by barbudor** [comment]
@AhmadNaeemBEL 
Your screen shot show that Content-Length is returned. But is that coming from the effective transaction between your ESP and the server ?
Can you capture the exchange using tcpdump/wireshark and check the presence of the header ?

**Reply by AhmadNaeemBEL** [reply]
No I just check the URL on Postman.
I didn't check if headers are coming from effective transaction between server and ESP.

Okay I will try to capture.

Thanks for the reply.

**Comment by sfromis** [comment, marked as answer]
My private OTA server only returns a few header fields. In addition to status, it is Date, Server, Content-Length and Content-Type. I'd not expect Server to be important, just found it clearer to identify the source. Are you using http? Encrypted https is not supported.

**Reply by AhmadNaeemBEL** [reply]
Yes I am using http.

But it is not working.

**Comment by ascillato** [comment]
TasmoAdmin ( https://github.com/reloxx13/TasmoAdmin ) uses OTAUrl and it is working fine.

As explained before, OTAUrl only works with http protocol. Https is not supported.

**Reply by AhmadNaeemBEL** [reply]
I am using http.

But it is not working.
